Public/Src/FrontEnd/Nuget/NugetResolver.cs

```diff
@@ -80,79 +80,85 @@ public override async Task<bool> InitResolverAsync(IResolverSettings resolverSet
             {
                 var cgManfiestGenerator = new NugetCgManifestGenerator(Context);
                 string generatedCgManifest = cgManfiestGenerator.GenerateCgManifestForPackages(maybePackages.Result);
-                string existingCgManifest = "INVALID";
 
-                if ( !Configuration.FrontEnd.GenerateCgManifestForNugets.IsValid &&
-                      Configuration.FrontEnd.ValidateCgManifestForNugets.IsValid )
+                if ( Configuration.FrontEnd.ValidateCgManifestForNugets.IsValid )
                 {
-                    // Validation of existing cgmainfest.json results in failure due to mismatch. Should fail the build in this case.
-                    try
+                    if (    Configuration.FrontEnd.GenerateCgManifestForNugets.IsValid
+                        &&  Configuration.FrontEnd.ValidateCgManifestForNugets.Equals(Configuration.FrontEnd.GenerateCgManifestForNugets))
                     {
-                        existingCgManifest = File.ReadAllText(Configuration.FrontEnd.ValidateCgManifestForNugets.ToString(Context.PathTable));
-                        FrontEndHost.Engine.RecordFrontEndFile(
-                            Configuration.FrontEnd.ValidateCgManifestForNugets,
-                            CGManifestResolverName);
+                        // Skip validation since the newly generated file will always be valid

```

Comment:
let's not have empty `if` body and merge this condition with the condition above